libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -858,19 +803,108 @@ bool HTTPClient::connect(void)
 
     _client->setTimeout(_tcpTimeout);
 
-    if(!_client->connect(_host.c_str(), _port)) {
-        DEBUG_HTTPCLIENT("[HTTP-Client] failed connect to %s:%u\n", _host.c_str(), _port);
+    String host = _uri->host;
```

Comment:
`host` can be a reference here?

Reply:
I don't believe so, its reassigned right below. ` host = _proxyUri->host;` If it were a reference like `String &host = _uri->host;` it would reassign the value of `_uri->host` which is not what is intended

Reply:
True :/ Using a pointer instead would avoid a useless allocation

Reply:
done